File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/client/block/Blocks.java
package pize.tests.voxelgame.client.block;

import pize.tests.voxelgame.Minecraft;
import pize.tests.voxelgame.client.block.vanilla.*;
import pize.tests.voxelgame.client.resources.GameResources;

import java.util.HashMap;
import java.util.Map;

public class Blocks{

    private final static Map<Byte, BlockProperties> fromID = new HashMap<>();
    
    public static BlockProperties fromID(byte blockID){
        return fromID.get(blockID);
    }
    
    public static BlockProperties register(BlockProperties properties){
        fromID.put(properties.getID(), properties);
        return properties;
    }

    public static void init(Minecraft session){
        final GameResources resources = session.getRenderer().getSession().getResources();
        for(BlockProperties blockProperties: fromID.values())
            blockProperties.load(resources);
    }
    
    public static final BlockProperties VOID_AIR    = register(new VoidAir   (-1 ));
    public static final BlockProperties AIR         = register(new Air       ( 0 ));
    public static final BlockProperties DIRT        = register(new Dirt      ( 1 ));
    public static final BlockProperties GRASS_BLOCK = register(new GrassBlock( 2 ));
    public static final BlockProperties STONE       = register(new Stone     ( 3 ));
    public static final BlockProperties GLASS       = register(new Glass     ( 4 ));
    public static final BlockProperties OAK_LOG     = register(new OakLog    ( 5 ));
    public static final BlockProperties LAMP        = register(new Lamp      ( 6 ));
    public static final BlockProperties GRASS       = register(new Grass     ( 7 ));
    public static final BlockProperties OAK_LEAVES  = register(new OakLeaves ( 8 ));
    public static final BlockProperties WATER       = register(new Water     ( 9 ));
    public static final BlockProperties SAND        = register(new Sand      ( 10));

}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/client/block/Blocks.kt
package pize.tests.voxelgame.client.block

import pize.tests.voxelgame.Minecraft
import pize.tests.voxelgame.client.block.vanilla.*

object Blocks {
    private val fromID: MutableMap<Byte, BlockProperties> = HashMap()
    fun fromID(blockID: Byte): BlockProperties? {
        return fromID[blockID]
    }

    fun register(properties: BlockProperties): BlockProperties {
        fromID[properties.id] = properties
        return properties
    }

    fun init(session: Minecraft) {
        val resources = session.renderer.session.resources
        for (blockProperties in fromID.values) blockProperties.load(resources)
    }

    val VOID_AIR = register(VoidAir(-1))
    val AIR = register(Air(0))
    val DIRT = register(Dirt(1))
    val GRASS_BLOCK = register(GrassBlock(2))
    val STONE = register(Stone(3))
    val GLASS = register(Glass(4))
    val OAK_LOG = register(OakLog(5))
    val LAMP = register(Lamp(6))
    val GRASS = register(Grass(7))
    val OAK_LEAVES = register(OakLeaves(8))
    val WATER = register(Water(9))
    val SAND = register(Sand(10))
}
